benchto-service/src/main/java/io/trino/benchto/service/model/Metric.java

```diff
@@ -0,0 +1,164 @@
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.MapKeyColumn;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Size;
+
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Objects;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.collect.Maps.newHashMap;
+import static javax.persistence.FetchType.EAGER;
+import static org.hibernate.annotations.CacheConcurrencyStrategy.TRANSACTIONAL;
+
+@Entity
+@Table(name = "metrics")
+public class Metric
+        implements Serializable
+{
+    @Id
+    @SequenceGenerator(name = "metrics_id_seq",
+            sequenceName = "metrics_id_seq",
+            allocationSize = 1)
+    @GeneratedValue(strategy = GenerationType.SEQUENCE,
+            generator = "metrics_id_seq")
+    @Column(name = "id")
+    @JsonIgnore
+    private long id;
+
+    @Size(min = 1, max = 64)
+    @Column(name = "name")
+    private String name;
+
+    @NotNull
+    @Enumerated(EnumType.STRING)
+    @Column(name = "unit")
+    private MeasurementUnit unit;
+
+    @Cache(usage = TRANSACTIONAL)
+    @BatchSize(size = 10)
+    @ElementCollection(fetch = EAGER)
+    @MapKeyColumn(name = "name")
+    @Column(name = "value")
+    @CollectionTable(name = "metric_attributes", joinColumns = @JoinColumn(name = "metric_id"))
+    private Map<String, String> attributes = newHashMap();
+
+    protected Metric()
+    {
+    }
+
+    public Metric(String name, MeasurementUnit unit, Map<String, String> attributes)
+    {
+        this.name = name;
+        this.unit = unit;
+        this.attributes = attributes;
+    }
+
+    public long getId()
+    {
+        return id;
+    }
+
+    public void setId(long id)
+    {
+        this.id = id;
+    }
+
+    public String getName()
+    {
+        return name;
+    }
+
+    public void setName(String name)
+    {
+        this.name = name;
+    }
+
+    public MeasurementUnit getUnit()
+    {
+        return unit;
+    }
+
+    public void setUnit(MeasurementUnit unit)
+    {
+        this.unit = unit;
+    }
+
+    public Map<String, String> getAttributes()
+    {
+        return attributes;
+    }
+
+    public void setAttributes(Map<String, String> attributes)
+    {
+        this.attributes = attributes;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Metric that = (Metric) o;
+        return Objects.equals(name, that.name)
+                && unit.equals(that.unit)
+                && attributes.equals(that.attributes);
```

Comment:
What if `attributes` is `null`? In a `hashcode` you check for it.

Reply:
Fixed, thanks!